gnd/src/main/java/com/google/android/gnd/ui/recorddetails/RecordDetailsViewModel.java

```diff
@@ -17,32 +17,124 @@
 package com.google.android.gnd.ui.recorddetails;
 
 import android.arch.lifecycle.LiveData;
-import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.LiveDataReactiveStreams;
+import android.util.Log;
+import android.view.View;
+
 import com.google.android.gnd.repository.DataRepository;
 import com.google.android.gnd.repository.Resource;
 import com.google.android.gnd.ui.common.AbstractViewModel;
+import com.google.android.gnd.vo.Form;
 import com.google.android.gnd.vo.Record;
+import com.google.common.collect.ImmutableList;
+
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.Flowable;
+import io.reactivex.subjects.BehaviorSubject;
+import io.reactivex.subjects.Subject;
+
 import javax.inject.Inject;
 
 public class RecordDetailsViewModel extends AbstractViewModel {
 
+  private static final String TAG = RecordDetailsViewModel.class.getSimpleName();
+
   private final DataRepository dataRepository;
-  private final MutableLiveData<Resource<Record>> record;
+  private final Subject<RecordDetailsFragmentArgs> recordDetailsRequest;
+  public final Flowable<Resource<Record>> recordStream;
+  public final LiveData<Resource<Record>> records;
+  public final LiveData<Integer> progressBarVisibility;
+  public final LiveData<String> toolbarTitle;
+  public final LiveData<String> toolbarSubtitle;
+  public final LiveData<String> formNameView;
 
   @Inject
   RecordDetailsViewModel(DataRepository dataRepository) {
     this.dataRepository = dataRepository;
-    this.record = new MutableLiveData<>();
+
+    this.recordDetailsRequest = BehaviorSubject.create();
+
+    this.recordStream =
+        recordDetailsRequest
+          .switchMap(a ->
```

Comment:
Nit: s/a/args/

Reply:
Sure thing, on this subject, is there a preferred guideline for naming arguments? I usually default to longer, descriptive names, *except* in lambdas--are we just following the google style guide? Maybe we should note this in CONTRIBUTING.md?

@gino-m